--- a/src/log/text_log.cc
+++ b/src/log/text_log.cc
@@ -31,6 +31,7 @@
 
 #include "text_log.h"
 
+#include <fcntl.h>
 #include <sys/stat.h>
 #include <time.h>
 
@@ -72,11 +73,14 @@
     if ( name && !strcasecmp(name, "stdout") )
     {
 #ifdef USE_STDLOG
-        FILE* stdlog = fdopen(STDLOG_FILENO, "w");
-        return stdlog ? stdlog : stdout;
-#else
-        return stdout;
+        // Check if the file handle exists, on musl, fdopen will just return the given handle
+        if ( fcntl(STDLOG_FILENO, F_GETFD) != -1 )
+        {
+            FILE* stdlog = fdopen(STDLOG_FILENO, "w");
+            return stdlog ? stdlog : stdout;
+        }
 #endif
+        return stdout;
     }
 
     return OpenAlertFile(name, is_critical);
--- a/src/pub_sub/CMakeLists.txt
+++ b/src/pub_sub/CMakeLists.txt
@@ -45,6 +45,7 @@
     ${PUB_SUB_INCLUDES}
     cip_events.cc
     detection_events.cc
+    dhcp_events.cc
     dns_events.cc
     eof_event.cc
     http_body_event.cc
--- /dev/null
+++ b/src/pub_sub/dhcp_events.cc
@@ -0,0 +1,15 @@
+
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "pub_sub/dhcp_events.h"
+
+namespace snort
+{
+
+// This defines the destructor inside the 'snort' namespace
+DHCPInfoEvent::~DHCPInfoEvent() = default;
+
+}
--- a/src/pub_sub/dhcp_events.h
+++ b/src/pub_sub/dhcp_events.h
@@ -21,6 +21,7 @@
 #define DHCP_EVENTS_H
 
 #include <cstring>
+#include "framework/data_bus.h"
 #include "pub_sub/appid_event_ids.h"
 
 #define DHCP_OP55_MAX_SIZE  64
@@ -29,9 +30,13 @@
 namespace snort
 {
 
-class DHCPInfoEvent : public snort::DataEvent
+class __attribute__((visibility("default"))) DHCPInfoEvent : public snort::DataEvent
+//class SO_PUBLIC DHCPInfoEvent : public snort::DataEvent
+//class DHCPInfoEvent : public snort::DataEvent
 {
 public:
+    virtual ~DHCPInfoEvent();
+
     DHCPInfoEvent(const snort::Packet* p, uint32_t ip_address, const uint8_t* eth,
         uint32_t subnet_mask, uint32_t lease_secs, uint32_t router) :
         pkt(p), ip_address(ip_address), subnet_mask(subnet_mask),
